test/integration/model/create.test.js

```diff
@@ -987,7 +987,15 @@ describe(Support.getTestDialectTeaser('Model'), () => {
       return UserNull.sync({ force: true }).then(() => {
         return UserNull.create({ username: 'foo2', smth: null }).catch(err => {
           expect(err).to.exist;
-          expect(err.get('smth')[0].path).to.equal('smth');
+
+          const smth1 = err.get('smth')[0] || {};
+
+          expect(smth1.path).to.equal('smth');
+          expect(smth1.type || smth1.origin).to.match(/notNull Violation/);
+
+          /*
+          // TODO: code below specifies different tests for different dialects but all the test are exactly the same.
+                   can we remove this or do we need to add tests to actually support different dialects?
```

Comment:
@sequelize/orm Does anyone know why? seems like this is really unnecessary

Reply:
my guess is that it was a consequence of the fact that the `ValidationErrorItem.type` property is an arbitrary string which can be set via the `ValidationErrorItem` constructor - something I consider bad-practice because it means you cannot rely on this property programmatically (much like error messages in general) - which is why I opted to implement `.origin` as a structured alternative to `.type`

*please note*: I intentionally left `.type` untouched (but marked as deprecated) so that this PR is BC ... meaning anyone who might, incorrectly ;-), be relying on `ValidationErrorItem.type` does not get bitten.

Reply:
Nope, no idea :)

Reply:
@iamjochem just remove this section, its in this state from at least 4 years